### PR TITLE
🎉New Source: Zenefits

### PR DESCRIPTION
**Pull request by makyash**
## What
*Describe what the change is solving* 
Hello Team, <br>
We at Pliancy, have created this python custom connector, that interacts with the API and pulls the data from the source. This connector helps in time tracking and is used by many companies like ours. Everything is explained properly on how to work this connector locally as well in the readme.md file.

*It helps to add screenshots if it affects the frontend.*

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*
Screen Shot 2022-07-08 at 3 50 40 PM

Screen Shot 2022-07-08 at 3 51 05 PM

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*
Screen Shot 2022-07-18 at 2 29 48 PM

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*
Screen Shot 2022-07-15 at 6 18 41 PM

</details>

### CODE REVIEW

**Comment by makyash** [comment]
here's my email:  yash.makwana@pliancy.com, do reach out to me for any questions.

**Comment by marcosmarxm** [comment]
@harshithmullapudi could you review this and publish without the credentials for now?

File: airbyte-integrations/connectors/source-zenefits/source_zenefits/schemas/TODO.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# TODO: Define your stream schemas
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenefits/source_zenefits/schemas/TODO.md:1]: Remove this file.

File: airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+        if next_page_url:
+            next_url = urlparse(next_page_url)
+            next_params = parse_qs(next_url.query)
+            return next_params
+
+        return None
+
+
+# Employee
+class people(ZenefitsStream):
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py:51]: Python Class are Capitalize.

File: airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+        params = {"limit": self.LIMIT}
+        if next_page_token:
+            params = next_page_token
+        return params
+
+    def request_headers(self, **kwargs) -> Mapping[str, Any]:
+        token = self.token
+        people_headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json", "Accept": "application/json"}
+        return people_headers
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py:37]:
```suggestion
    def request_headers(self, **kwargs) -> Mapping[str, Any]:
        return {"Authorization": f"Bearer {self.token}", "Content-Type": "application/json", "Accept": "application/json"}
```

File: airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+        self,
+        response: requests.Response,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> Iterable[Mapping]:
+
+        employments_response = response.json().get("data")
+        eresponse = employments_response.get("data")
+        return eresponse
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py:90]: Move this to parent stream as

```python
def parse_response(...):
    return response.json().get("data", {}).get("data")

File: airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py [hunk 5]
```diff
@@ -0,0 +1,306 @@
...
+        token = config["token"]
+        headers = {"Authorization": f"Bearer {token}", "Content-Type": "application/json", "Accept": "application/json"}
+        url = "https://api.zenefits.com/core/people"
+
+        try:
+            session = requests.get(url, headers=headers)
+            session.raise_for_status()
+            return True, None
+        except requests.exceptions.RequestException as e:
+            return False, e
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenefits/source_zenefits/source.py:290]: Check how other connectors made this implementation for `check_connection`.

File: airbyte-integrations/connectors/source-zenefits/source_zenefits/spec.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "title": "Zenefits Integration Spec",
+    "type": "object",
+    "required": ["token"],
+    "additionalProperties": false,
+    "properties": {
+      "token":{
+        "title":"token",
+        "type": "string",
+        "description": "describe me"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-zenefits/source_zenefits/spec.json:13]: update the `describe me`.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
I made a few comments, sorry the long delay to review it.

**Comment by makyash** [comment]
@marcosmarxm  I have made the necessary changes you suggested, the only thing with check_connection is I have a Bearer Token and I have to work surround that. 
I ran source acceptance test again after making the changes to help speed up the process, you can find the test screenshot below

Screen Shot 2022-08-18 at 10 31 00 AM

**Comment by harshithmullapudi** [comment]
Hey @makyash is it possible to give me access to push a commit over this repo. I have some fixes to make and a documentation to add. If not do you mind doing these

1. Add a new record in source_definition file just after zendesk
```
- name: Zenefits
  sourceDefinitionId: 8baba53d-2fe3-4e33-bc85-210d0eb62884
  dockerRepository: airbyte/source-zenefits
  dockerImageTag: 0.1.0
  documentationUrl: https://docs.airbyte.io/integrations/sources/zenefits
  icon: zenefits.svg
  sourceType: api
  releaseStage: alpha
```
2. Create a new file in docs/integrations/sources and name it zenefits.md and contents attached below
3. Remove sample_files folder and then remove all TODO.md files
[zenefits.md](https://github.com/airbytehq/airbyte/files/9413978/zenefits.md)

4. Run `./gradlew format`

**Comment by makyash** [comment]
@harshithmullapudi let me know, if the changes were accurate or else I will give you the access to push  a commit into this repo.

**Comment by harshithmullapudi** [comment]
Sure checking

**Comment by harshithmullapudi** [comment]
LGTM

**Review by harshithmullapudi — APPROVED** [review]

**Comment by harshithmullapudi** [comment]
/publish connector=connectors/source-zenefits run-tests=false

**Pull request merged**